src/components/chips/js/chipsController.js

```diff
@@ -0,0 +1,86 @@
+(function () {
+  'use strict';
+  angular
+      .module('material.components.chips')
+      .controller('MdChipsCtrl', MdChipsCtrl);
+
+  function MdChipsCtrl ($scope, $element, $q, $mdUtil, $mdConstant, $log) {
+    //-- private variables
+    var self = this;
+    this.$mdConstant = $mdConstant;
+
+    /** @type {!angular.NgModelController} */
+    this.ngModelCtrl = null;
+
+    /** @type {Array.<Object>} */
+    this.items = [];
+
+    //-- public variables
+    self.scope = $scope;
+
+    return self.init();
+
+    // How to update the model:
+    // self.ngModelCtrl.$setViewValue(new Date(Number(this.dataset.timestamp)));
+    // self.ngModelCtrl.$render();
+
+  }
+
+  /** 
+   * The model used by the input element.
+   * @type {!String}
+   */
+  MdChipsCtrl.prototype.itemBuffer = '';
```

Comment:
This is probably a side-effect of me coming from projects that heavily
employed Closure where having public (and private) members on the prototype
is the rule.

On Tue, Mar 24, 2015 at 10:23 AM Jeremy Elbourn <notifications@github.com>
wrote:

> In src/components/chips/js/chipsController.js
> <https://github.com/angular/material/pull/1981#discussion_r27051353>:
>
> > +    //-- public variables
> > +    self.scope = $scope;
> > +
> > +    return self.init();
> > +
> > +    // How to update the model:
> > +    // self.ngModelCtrl.$setViewValue(new Date(Number(this.dataset.timestamp)));
> > +    // self.ngModelCtrl.$render();
> > +
> > +  }
> > +
> > +  /**
> > +   * The model used by the input element.
> > +   * @type {!String}
> > +   */
> > +  MdChipsCtrl.prototype.itemBuffer = '';
>
> Yeah, you're right, it won't cause any issues, but it has no reason to be
> on the prototype either way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/material/pull/1981/files#r27051353>.
>